airbyte-integrations/connectors/source-smartsheets/source_smartsheets/spec.json

```diff
@@ -4,15 +4,68 @@
     "$schema": "http://json-schema.org/draft-07/schema#",
     "title": "Smartsheets Source Spec",
     "type": "object",
-    "required": ["access_token", "spreadsheet_id"],
+    "required": ["credentials", "spreadsheet_id"],
     "additionalProperties": true,
     "properties": {
-      "access_token": {
-        "title": "Access Token",
-        "description": "The access token to use for accessing your data from Smartsheets. This access token must be generated by a user with at least read access to the data you'd like to replicate. Generate an access token in the Smartsheets main menu by clicking Account > Apps & Integrations > API Access. See the <a href=\"https://docs.airbyte.com/integrations/sources/amplitude#setup-guide\">setup guide</a> for information on how to obtain this token.",
-        "type": "string",
+      "credentials": {
+        "title": "Authorization Method",
+        "type": "object",
         "order": 0,
-        "airbyte_secret": true
+        "oneOf": [
+          {
+            "type": "object",
+            "title": "OAuth2.0",
+            "required": ["client_id", "client_secret", "refresh_token", "access_token", "token_expiry_date"],
+            "properties": {
+              "auth_type": {
+                "type": "string",
+                "const": "oauth2.0"
+              },
+              "client_id": {
+                "type": "string",
+                "description": "The API ID of the SmartSheets developer application.",
+                "airbyte_secret": true
+              },
+              "client_secret": {
+                "type": "string",
+                "description": "The API Secret the SmartSheets developer application.",
+                "airbyte_secret": true
+              },
+              "access_token": {
+                "type": "string",
+                "description": "Access Token for making authenticated requests.",
+                "airbyte_secret": true
+              },
+              "token_expiry_date": {
+                "type": "string",
+                "description": "The date-time when the access token should be refreshed.",
+                "format": "date-time"
+              },
+              "refresh_token": {
+                "type": "string",
+                "description": "The key to refresh the expired access_token.",
+                "airbyte_secret": true
+              }
+            }
+          },
+          {
+            "title": "Private Token",
```

Comment:
It looks like Smartsheets calls these "API Access Token"s https://help.smartsheet.com/articles/2482389-generate-API-key

Reply:
updated